src/cluster/src/error.rs

```diff
@@ -45,8 +45,8 @@ pub enum K8InstallError {
     #[error("Failed to perform one or more pre-checks")]
     PrecheckErrored(CheckResults),
     /// Failed to update Fluvio cluster
-    #[error("Failed to update Fluvio")]
-    FailedUpdate,
+    #[error("Failed to witness expected Fluvio cluster version: {0}")]
+    FailedVersionTimeout(String),
```

Comment:
Isn't this same as SCServiceTimeout?  I would expect error related to helm upgrade failure like (existing install version, etc)

Reply:
I think SC Service timeout is about waiting for the endpoint to be available, and the Failed Version timeout is about waiting to see a Fluvio cluster with the specifically requested version. This is what the `upgrade` process does, because when you begin upgrading, there is already an SC up and available, but it does not indicate that the upgrade has succeeded, because you would expect the old SC to go down and to get a new one up that reports the correct version.

Reply:
I can look at cleaning this up when I refactor, but this at least gets the job done

Reply:
FailedVersionTimeout is confusing and misleading.  There could be multiple reason:
1:  Helm installation failure.
2.  Upgraded failed for whatever reason, SC doesn't come up?  
3.  Network failure
4.  SC come up but somehow version is different.

For 2&3, `SCServiceTimeout` can be used.

Reply:
I have been playing with this for a little bit and it seems like I won't be able to actually merge `wait_for_fluvio_version` and `wait_for_sc_service` because `create_managed_spu_group` needs to run in between those calls. The flows involved here aren't quite obvious because similar code is being used for both the initial-install path and the upgrade path. I am hoping to untangle these a bit in the future.

So here's where we're at:

- Fluvio install worked just fine before I added upgrade.
- As part of upgrade, I needed to be able to look at the "platform version" reported by the cluster, because simply observing an SC and SPU standing is not enough to know that they were actually upgraded to the specified version.
- So `wait_for_fluvio_version` is really only needed for upgrade use-case.

My current thinking is that I can reword the error message to indicate that something went wrong in the upgrade process, but I don't know how to be any more specific than "the upgrade failed" if we are not going to talk about the fact that the version requests did not see the new version of the cluster appear.

Reply:
`create_managed_spu_group`  is optional

Reply:
new error message is fine

Reply:
Alright, I've pushed a new error message which is `FailedClusterUpgrade` which prints as "Failed to upgrade cluster to version X"